Error on deploy when moving files on rinkeby

Seen on the rinekby backend:

    [ERROR] dshop.routes.shops: TypeError: Cannot read property 'path' of undefined
        at Promise (/app/backend/routes/shops.js:544:21)
        at new Promise (<anonymous>)
        at form.parse (/app/backend/routes/shops.js:543:17)
        at IncomingForm.<anonymous> (/app/node_modules/formidable/lib/incoming_form.js:107:9)
        at IncomingForm.emit (events.js:198:13)
        at IncomingForm.EventEmitter.emit (domain.js:448:20)
        at IncomingForm._maybeEnd (/app/node_modules/formidable/lib/incoming_form.js:557:8)
        at MultipartParser.parser.onEnd (/app/node_modules/formidable/lib/incoming_form.js:426:10)
        at callback (/app/node_modules/formidable/lib/multipart_parser.js:102:31)
        at MultipartParser.write (/app/node_modules/formidable/lib/multipart_parser.js:268:15)
        at IncomingForm.write (/app/node_modules/formidable/lib/incoming_form.js:159:34)
        at IncomingMessage.<anonymous> (/app/node_modules/formidable/lib/incoming_form.js:125:12)
        at IncomingMessage.emit (events.js:198:13)
        at IncomingMessage.EventEmitter.emit (domain.js:448:20)
        at IncomingMessage.Readable.read (_stream_readable.js:505:10)
        at flow (_stream_readable.js:974:34)
        at resume_ (_stream_readable.js:955:3)

--- a/backend/routes/shops.js
+++ b/backend/routes/shops.js
@@ -498,8 +498,15 @@ module.exports = function (app) {
         try {
           for (const file of allFiles) {
             await new Promise((resolve, reject) => {
-              mv(file.path, `${uploadDir}/${file.name}`, (err) => {
-                return err ? reject(err) : resolve()
+              fs.stat(file.path, (err, fstat) => {
+                if (err) return reject(err)
+                if (!fstat.isFile()) {
+                  return reject(`${file.path} does not exist or is not a file`)
+                }
+
+                mv(file.path, `${uploadDir}/${file.name}`, (err) => {
+                  return err ? reject(err) : resolve()
+                })
               })
             })
           }
@@ -535,14 +542,24 @@ module.exports = function (app) {
         }
 
         const { file } = files
+        if (!file) {
+          return res.json({ success: false, reason: 'no-file' })
+        }
         if (Array.isArray(file)) {
           return res.json({ success: false, reason: 'too-many-files' })
         }
 
         try {
           await new Promise((resolve, reject) => {
-            mv(file.path, `${uploadDir}/${file.name}`, (err) => {
-              return err ? reject(err) : resolve()
+            fs.stat(file.path, (err, fstat) => {
+              if (err) return reject(err)
+              if (!fstat.isFile()) {
+                return reject(`${file.path} does not exist or is not a file`)
+              }
+
+              mv(file.path, `${uploadDir}/${file.name}`, (err) => {
+                return err ? reject(err) : resolve()
+              })
             })
           })
 

